lms/djangoapps/edraak_forus/tests/tests.py

```diff
@@ -0,0 +1,388 @@
+# pylint: disable=missing-docstring
+import json
+from datetime import datetime, timedelta
+from urlparse import urlparse
+
+import ddt
+import pytz
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect
+from django.test import TestCase, override_settings
+from mako.filters import html_escape
+from mock import patch, Mock
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+from hashlib import sha256
+import hmac
+
+from edraak_forus.helpers import ValidateForusParams, ordered_hmac_keys, calculate_hmac
+from edraak_forus.models import ForusProfile
+from edraak_forus.tests.helpers import build_forus_params
+
+NEXT_WEEK = datetime.now(pytz.UTC) + timedelta(days=7)
+PAST_WEEK = datetime.now(pytz.UTC) - timedelta(days=7)
+NEXT_MONTH = datetime.now(pytz.UTC) + timedelta(days=30)
+YESTERDAY = datetime.now(pytz.UTC) - timedelta(days=1)
+
+
+class SettingsTest(TestCase):
+    """
+    Sanity checks for the environment settings
+    """
+    def test_if_enabled(self):
+        """
+        Ensure the app is enabled
+        """
+        self.assertIn('edraak_forus', settings.INSTALLED_APPS, 'Forus app should be enabled in tests')
+
+    def test_url_in_settings(self):
+        self.assertEqual(settings.FORUS_BASE_URL, 'http://forus.jo')
+
+    def test_secret_key_in_settings(self):
+        self.assertEqual(settings.FORUS_AUTH_SECRET_KEY, 'dummy_auth_secret_key')
+
+
+@patch('edraak_forus.helpers.calculate_hmac', Mock(return_value='dummy_hmac'))
+class ForusAuthTest(ModuleStoreTestCase):
+    """
+    Test the ForUs auth.
+    """
+
+    def setUp(self):
+        super(ForusAuthTest, self).setUp()
+        self.course = CourseFactory.create(
+            enrollment_start=PAST_WEEK,
+            start=NEXT_WEEK,
+        )
+
+        self.user_email = 'forus.user@example.com'
+
+        self.auth_url = reverse('forus_v1_auth')
+        self.register_url = reverse('forus_v1_reg_api')
+        self.course_root_url = '/courses/{}/info'.format(self.course.id)
+        self.dashboard_url = reverse('dashboard')
+
+    # pylint: disable=invalid-name
+    def _assertLoggedIn(self, msg_prefix=None):
+        res_dashboard = self.client.get(self.dashboard_url)
+        ## testing dashboard url to see if user is logged in
+        self.assertContains(res_dashboard, 'dashboard-main', msg_prefix=msg_prefix)
+
+    # pylint: disable=invalid-name
+    def _assertLoggedOut(self):
+        res_dashboard = self.client.get(self.dashboard_url)
+        self.assertEquals(res_dashboard.status_code, 302, 'User is not logged out.')
```

Comment:
Just use `self.assertRedirects()` looks like it has been invented later on :rofl: